docs/proposals/job_controller.md

```diff
@@ -0,0 +1,191 @@
+# Job Controller
+
+## Abstract
+A proposal for implementing a new controller - Job controller, which will be responsible
+for managing pods that require to run:
+* once on demand or at a specified point in time
+* repeatedly at a specified point in time.
+
+Several exiting issues were already created regarding that particular subject:
+* Distributed CRON jobs [#2156](https://github.com/GoogleCloudPlatform/kubernetes/issues/2156)
+* Job Controller [#1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1624)
+
+
+## Use Cases
+1. Ability to have a controller that will create a run-once pod.
+1. Ability to have a controller that will have an option to specify reaction on pod’s success/failure.
+1. Ability to specify number of restarts upon pod failure.
+1. Ability to specify maximum running time for a pod.
+1. Ability to create a repetitive job, eg. database backup, sending emails.
+1. Ability to create one time job to be run at a specified time, eg. system analysis.
+1. Ability to create a chain of jobs dependent one on another.
+
+
+## Motivation
+The main goal is to provide a new pod controller type, called Job, with the following
+characteristics:
+* Be able to start of one or several pods tracked as a single entity representing
+a one execution of this job. Final job execution status will reflect the completion
+status of all pods started for this run.
+* Be able to trigger a new job execution either on demand (triggered through server
+API by an external agent or by another subsystem), or periodically based on a time-based
+schedule defined in job specification part (using ISO8601 syntax notation).
+* Be able to restart a limited number of time failing containers of pods created
+by one job execution (using "Restart on failure" logic extended by a maximal restart count).
+* Be able to limit the execution time of pods created by a job.
```

Comment:
So you're saying we should do both or just the job part? If the job, how we should limit the time of a job consisting of multiple pods, does it mean that each of the pods should run X, given the X is the job limit? I was hoping to use here the same pattern we have in [deployments](https://github.com/openshift/origin/blob/master/pkg/deploy/api/types.go#L36).